### PR TITLE
[Script] setup stable x environment

### PR DESCRIPTION
**Pull request by fleupold**
This PR creates the first StableX interaction script.

For that, moves the existing scripts that are solely used for the snapp version to a subfolder and creates a new stablex specific script folder with the first `setup_environment` script.\

### Test Plan
run migration and `npx truffle exec scripts/stablex/setup_environment.js`. Then use `truffle console` to check that allowances and balances have been correctly set.

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Cool stuff! good first script and I like the move of snapp specific into sub directory. It appears there is something wrong with the coverage reporter now... Have the migration scripts changed?

File: scripts/stablex/setup_environment.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    const numAccounts = arguments[0] || 3
+    const numTokens = arguments[1] || 3
+    const instance = await StablecoinConverter.deployed()
+    const accounts = await web3.eth.getAccounts()
+    console.log(`Beginning environment setup with ${numAccounts} accounts and ${numTokens} tokens`)
+
+    const amount = "300000000000000000000"
+
+    // Create and register tokens (feeToken is already registered)
+    const tokens = [await ERC20Mintable.deployed()]
```
Comment (josojo) [inline scripts/stablex/setup_environment.js:20]: Maybe we can here make sure that this is really the fee token:
await ERC20Mintable.at(await StablecoinConverter.feeToken()).deployed()]

**Review by josojo — COMMENTED** [review]
I am not sure, whether requiring the feeToken to implement the mint function is really a good idea. 

For example, for the GNO token, we don't have this functionality. Just all tokens are immediately credited to account[0] at deployment.

While it is helpful now, I guess we have to change it later, if we wanna use this script in Rinkeby.

**Comment by fleupold** [comment]
> While it is helpful now, I guess we have to change it later, if we wanna use this script in Rinkeby.

The setup_environment script is not going to be used on a production or testnet deployment. It relies 100% on mintable tokens (if not for the fee token, for all other traded and added tokens). All it does is generate available balance for the people participating with the system.

In a real setup, those balances will come from actually buying tokens on another exchange.

**Review by josojo — APPROVED** [review]
Alright. Yeah, that sounds good. We will only use it for internal e2e tests

**Review by anxolin — COMMENTED** [review]
Arrived late, it was merged super fast.

most of the changes seems formatting that is not related to the PR.

I understand this script is for the e2e test. I think it's good

**Pull request merged**